File: task2/task2.js
function isPrime(n){
	var sqrtN = Math.round(Math.sqrt(n)) + 1;

    if(n==1) return false;
    for(var d=2; d <= sqrtN; d++){
        if(n%d==0) return false;
    };
    return true;
};

var n = 600851475143;
var prime = 2;

while (n != 1) {
	if (n % prime == 0){
		print(n,"/",prime,"=",n=n/prime);
	}
	else {
		while (!isPrime(++prime)){};
	}
}

print("Max prime divider:", prime);

File: task2/task2.java
import java.io.*;

class task2 {
	public static void main(String args[ ]) {
		long n = 600851475143L;
		int prime = 2;
		task2 IP = new task2();

		System.out.println("Start search max prime for " + n + "...");

		while (n != 1){
			if (n%prime == 0)
			    System.out.println(n + " / " + prime + " = " + (n=n/prime));
			else
			    while(!IP.isPrime(++prime)){};
		};

		System.out.println("Max prime = " + prime);

	}

	private boolean isPrime(long n){
		long d;

		for (d=2; d*d <= n; d++)
			if (n%d == 0) return false;

		return true;
	}
}